server/src/main/java/org/elasticsearch/index/mapper/MapperService.java

```diff
@@ -497,6 +500,7 @@ static void validateTypeName(String type) {
             ContextMapping.validateContextPaths(indexSettings.getIndexVersionCreated(), fieldMappers, fieldTypes::get);
 
             if (reason == MergeReason.MAPPING_UPDATE) {
+                checkCompletionContextsLimit(fieldMappers);
```

Comment:
I think it would be easier to perform the check directly in CompletionFieldMapper#Builder#build since you have all the needed informations there. We should also not throw an error on old indices (created before 780) since they were created before this breaking change. You can check the created version of the index in the `CompletionFieldMapper` with `BuilderContext#indexCreatedVersion` and throw an error if the index has been created on or after 8.0 and a deprecation warning otherwise.